src/sentry/api/endpoints/project_codeowners.py

```diff
@@ -61,6 +63,16 @@ def validate(self, attrs: Mapping[str, Any]) -> Mapping[str, Any]:
         if not attrs.get("raw", "").strip():
             return attrs
 
+        # We want to limit `raw` to a reasonable length, so that people don't end up with values
+        # that are several megabytes large. To not break this functionality for existing customers
+        # we temporarily allow rows that already exceed this limit to still be updated.
+        # We do something similar with ProjectOwnership at the API level.
+        existing_raw = self.instance.raw or ""
+        if len(attrs["raw"]) > MAX_RAW_LENGTH and len(existing_raw) <= MAX_RAW_LENGTH:
+            raise serializers.ValidationError(
+                {"raw": f"Raw needs to be <= {MAX_RAW_LENGTH} characters in length"}
+            )
+
```

Comment:
LGTM but is there a plan for removing this 'temporary' permission?

Reply:
@leeandher yea probs something PM and SE's will communicate to those customers.